### PR TITLE
Fix WorkingDir of Driver.py

### LINKED ISSUE #1687: [DEFECT] Title Here
--------
Defect Description
--------

**Describe the defect**

##### What did you expect to see happen?

follow up of conversation https://github.com/idaholab/raven/pull/1677#discussion_r732397201

The working directory, if not specified via abs path, should always refer to the location of the input file 

##### What did you see instead?

a workingDir is generated if the input file is invoked from a different directory. This causes that the external model must
reside in this "newly generated working dir" but the results are stored in the working director at the location of the input file 

##### Do you have a suggested fix for the development team?

**Describe how to Reproduce**
1. 
2. 
3. 
4. 

**Screenshots and Input Files**
Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

**Platform (please complete the following information):**
 - Version: [e.g. 22]
 - Dependencies Installation: [CONDA or PIP]

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [ ] 1. Is it tagged with a type: defect or task?
- [ ] 2. Is it tagged with a priority: critical, normal or minor?
- [ ] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [ ] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [ ] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [ ] 1. If the issue is a defect, is the defect fixed?
- [ ] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [ ] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [ ] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [ ] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by dylanjm**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)

#1033
close #1687 
##### What are the significant changes in functionality due to this change request?

No significant changes. RAVEN will now adjust its working directory to be wherever the input file is pointed. 

For example, before this would break raven:

```
cd raven
./raven_framework tests/framework/user_guide/ravenTutorial/singleRun.xml
```

Now it doesn't. 

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: ravenframework/Driver.py [hunk 1]
```diff
@@ -141,6 +141,10 @@ def main(checkLibraries):
 
     # call the function to load the external xml files into the input tree
     cwd = os.path.dirname(os.path.abspath(inputFile))
+    # If a user chooses to run RAVEN outside of the directory containing the
+    # xml input file, then we should should change the python working directory
+    # to the directory containing the input file to avoid confusion.
+    os.chdir(cwd)
```
Comment (wangcj05) [inline ravenframework/Driver.py:147]: I think this change just solve part of the problem. Take the test "raven/tests/framework/test_iostep_load.xml" for example (see the following attached screenshot), the files provided by Files using relative path, which I think currently based on the WorkingDir provided by the RunInfo. I think the confusion here is we do not have a consistent way to explain and use the path. The issue #1687 suggest we use the input file folder as the base directory.  The following is my suggestion: I think the default behavior is use the input file folder as the base directory and search the files, if the files do not exist, then using the WorkingDir listed in RunInfo as the base directory, if the files are still not found, then use the running directory as the base directory. After trying search all these places, error will be raise if we can not find these files. @dylanjm @Jimmy-INL @joshua-cogliati-inl @PaulTalbot-INL @mandd please let me know what do you think.  

image

**Comment by wangcj05** [comment]
@dylanjm Do you have a change to resolve this issue?

**Comment by dylanjm** [comment]
@wangcj05 I'm having trouble finding where we read files in the code that still has access to the RunInfo node.

**Comment by wangcj05** [comment]
@dylanjm the path in the Code.py is assigned in initialize method, the 'subSubDirectory', is it the place that you are looking for?

image

**Comment by dylanjm** [comment]
@wangcj05 This could be the place I need to add it. My only question is are there files that won't go into Models that we need to check different directories for? Seems like this code is only checking the files that are passed as inputs to a Models block.

**Comment by wangcj05** [comment]
@dylanjm do you by chance have time to fix this issue?

**Comment by dylanjm** [comment]
> @dylanjm do you by chance have time to fix this issue?

@wangcj05 Ahh, yes. This PR had skipped my mind. I'll tackle it right now.

**Comment by dylanjm** [comment]
@wangcj05 A few things to note:

1. This change fixes the error you were running into with `raven/tests/framework/test_iostep_load.xml` You can run this test from `raven/tests` or `raven/tests/framework` and it will pass. 
2. The GA test was not passing because the External Model path was not relative to the working directory specified in the input file. It's also weird to name the working directory the same as another directory you want to put the external model in. We should encourage users to place the file in a premade working directory next to the input file or have them specify the relative path from the input file. 
3. There is no central place in RAVEN (that I can find) that would allow us to check for the existence of files in multiple locations. We also allow users to specify files that don't even exist yet. 

This fix allows users to run RAVEN from whichever directory they'd like but keeps the expected output the same no matter what. As long as users specify their files relative to the working directory.

I may be missing something, so I'm open to any feedback you have on the matter.

**Review by wangcj05 — APPROVED** [review]
PR looks good.

**Comment by wangcj05** [comment]
Checklist is good, and tests are green. PR can be merged. @dylanjm Thanks for your contribution.

**Pull request merged**